src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -161,25 +168,63 @@ internal static BindingDataResolver GetAppSettingResolver(string originalValue,
             return (newAttr, bindingData) => resolvedValue;
         }
 
+        // Run validition. This needs to be run at different stages. 
+        // In general, run as early as possible. If there are { } tokens, then we can't run until runtime.
+        // But if there are no { }, we can run statically. 
+        // If there's no [AutoResolve], [AppSettings], then we can run immediately.
+        private static void RunValidateAttributes(object value, PropertyInfo propInfo)
+        {
+            var attrs = propInfo.GetCustomAttributes<ValidationAttribute>();
+
+            foreach (var attr in attrs)
+            {
+                try
+                {
+                    attr.Validate(value, propInfo.Name);
+                }
+                catch (Exception e)
+                {
+                    throw new InvalidOperationException($"'{propInfo.Name}' can't be '{value}': {e.Message}");
```

Comment:
This "Not logging the value if also annotated with [AppSetting] might be slightly better." is a good compromise

---
In reply to: [123106464](https://github.com/Azure/azure-webjobs-sdk/pull/1188#discussion_r123106464) [](ancestors = 123106464)